star-rx72n-firmware/lib/rx_hal/src/rx_gptw.c

```
@@ -687,35 +687,189 @@ rx_err_t rx_gptw_stop(const rx_gptw_channel_t channel)
-  err = rx_gptw_enable_output(channel, k_gptw_output_b, false);
+
+  /* Calculate period once (assumes same frequency for all) */
+  err = internal_calculate_period(config->frequency_hz, config->wave_mode, &period);
   if (err != k_rx_ok) {
     return err;
   }
 
-  /* Mark as uninitialized */
-  s_gptw_initialized[channel] = false;
-  s_gptw_period[channel]      = k_gptw_period_zero;
+  const bool is_triangle = internal_is_triangle_mode(config->wave_mode);
```

Clang (14.0.6)

[warning] variable 'is_triangle' is not initialized

(cppcoreguidelines-init-variables)